pkg/products/rhsso/reconciler.go

```diff
@@ -124,31 +123,49 @@ func (r *Reconciler) Reconcile(ctx context.Context, installation *integreatlyv1a
 				return phase, err
 			}
 		}
-
+		_, err = resources.GetNS(ctx, r.Config.GetOperatorNamespace(), serverClient)
+		if !k8serr.IsNotFound(err) {
+			phase, err := resources.RemoveNamespace(ctx, installation, serverClient, r.Config.GetOperatorNamespace())
+			if err != nil || phase != integreatlyv1alpha1.PhaseCompleted {
+				return phase, err
+			}
+		}
 		err = resources.RemoveOauthClient(ctx, installation, serverClient, r.oauthv1Client, r.getOAuthClientName())
 		if err != nil {
 			return integreatlyv1alpha1.PhaseFailed, err
 		}
-		return integreatlyv1alpha1.PhaseCompleted, nil
+
+		//if both namespaces are deleted, return complete
+		_, operatorNSErr := resources.GetNS(ctx, r.Config.GetOperatorNamespace(), serverClient)
+		_, nsErr := resources.GetNS(ctx, r.Config.GetNamespace(), serverClient)
+		if k8serr.IsNotFound(operatorNSErr) && k8serr.IsNotFound(nsErr) {
+			return integreatlyv1alpha1.PhaseCompleted, nil
+		}
+		return integreatlyv1alpha1.PhaseInProgress, nil
```

Comment:
I think operator will get to this point only if the namespaces are gone, so this couple of lines above are not necessary, and we can leave just `return integreatlyv1alpha1.PhaseCompleted, nil` as it was before.
But if my assumption is incorrect, and there was a bug that caused this change, then we perhaps need to do it in all other products as well?

Reply:
I thought that too, but I was wrong, these reconcilers were reporting complete too early, and not removing both namespaces.

Reply:
@philbrookes ok, I think it could be attributed to the last condition here:
https://github.com/integr8ly/integreatly-operator/blob/9e160fdd416762b7a32667539b650f1a93b4802e/pkg/resources/finalizers.go#L44-L51

I think an "admin" permissions for our SA are removed before the NS is gone, and with that^ logic it gives a "false possitive".
To fix this, the `RemoveNamespace` function should ignore IsForbidden errors, and return PhaseCompleted only when GetNS return NotFound error. So something like this:
```
func RemoveNamespace(ctx context.Context, inst *integreatlyv1alpha1.Installation, client k8sclient.Client, namespace string) (integreatlyv1alpha1.StatusPhase, error) {
	ns, err := GetNS(ctx, namespace, client)
	if err != nil {
		if k8serr.IsNotFound(err)
			return integreatlyv1alpha1.PhaseCompleted, nil
		}
		// Since we are using ProjectRequests and limited permissions,
		// request can return "forbidden" error even when Namespace simply doesn't exist
		if !k8serr.IsForbidden(err) {
			logrus.Error("Error getting a namespace", err)
			return integreatlyv1alpha1.PhaseFailed, err
		}
	} else {
		if ns.Status.Phase == corev1.NamespaceTerminating {
	  		return integreatlyv1alpha1.PhaseInProgress, nil
		}
        }
...
```
Does that make sense?

Reply:
My code above makes it rather complicated.
Perhaps in the RemoveNamespace we should just be checking for the existence of Project resource, instead of `GetNS`. Maybe that would be more reliable and readable?

Reply:
I'll add something to tech debt for this.